### PR TITLE
798 Add Horovod distributed training example

### LINKED ISSUE #798: Add distributed training example with Horovod
**Is your feature request related to a problem? Please describe.**
Currently, we already have distributed training and evaluation examples based on PyTorch native APIs.
As Horovod is very popular in the distributed training area, add another example based on Horovod.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #798 .

### Description
This PR implemented a distributed training example based on Horovod APIs.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by Nic-Ma** [comment]
I tested locally on 1 machine with 2 GPUs, it trained as the expectation.
Need to set up SSH in dockers and test multi-nodes.
Thanks.

**Comment by Nic-Ma** [comment]
Tested with 2 distributed machines, the training process is faster than PyTorch native API as expected:
Overall training speed for this example is **46s**, while PyTorch native launch is **60s**, PyTorch MP method is **79s**.
And it's easier to launch, only need to run command on master node, no need to put program and dataset on other nodes.

Thanks.

File: examples/distributed_training/unet_training_horovod.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    ToTensord,
+)
+from monai.data import create_test_image_3d, Dataset, DataLoader
+
+
+def train(args):
+    # initialize Horovod library
+    hvd.init()
+    # Horovod limits CPU threads to be used per worker
+    torch.set_num_threads(1)
```
Comment (wyli) [inline examples/distributed_training/unet_training_horovod.py:73]: is this needed:
https://github.com/horovod/horovod/blob/34baedf1257bbdc5de15e54723d6d7bc21409a87/examples/pytorch_mnist.py#L126-L130

Comment (Nic-Ma) [reply examples/distributed_training/unet_training_horovod.py:74]: I tested both methods(num_worker=0 or 1), no obvious difference.
Anyway, I already updated to align with Horovod example in the latest commit.
Thanks.

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Thanks for your review.
I added test env information to the doc-string.
But please note that the test speed is significantly affected by network connection.
I tested with 2 machines at different floors of our office building, the speed is quite different when testing at different time.
So to compare the accurate speed benchmark, we need to do in NGC with the "real tutorial" later.
And BTW, I misunderstood a thing yesterday, we still need to put the program and data on other nodes with same path.

Thanks.

**Pull request merged**